src/zenml/integrations/langchain/materializers/openai_embedding_materializer.py

```diff
@@ -13,130 +13,21 @@
 #  permissions and limitations under the License.
 """Implementation of the Langchain OpenAI embedding materializer."""
 
-import os
-import pickle
 import sys
-from typing import TYPE_CHECKING, Any, Type, cast
-
-import pkg_resources
+from typing import TYPE_CHECKING, Any, ClassVar, Tuple, Type
 
 from zenml.enums import ArtifactType
-from zenml.environment import Environment
-from zenml.io import fileio
-from zenml.logger import get_logger
-from zenml.materializers.base_materializer import BaseMaterializer
-from zenml.utils.io_utils import (
-    read_file_contents_as_string,
-    write_file_contents_as_string,
-)
+from zenml.materializers.default_materializer import DefaultMaterializer
 
 if TYPE_CHECKING and sys.version_info < (3, 8):
     OpenAIEmbeddings = Any
 else:
     from langchain.embeddings import OpenAIEmbeddings
 
-DEFAULT_PICKLE_FILENAME = "embedding.pkl"
-DEFAULT_PYTHON_VERSION_FILENAME = "python_version.txt"
-DEFAULT_LANGCHAIN_VERSION_FILENAME = "langchain_version.txt"
-LANGCHAIN_PACKAGE_NAME = "langchain"
-
-logger = get_logger(__name__)
 
-
-class LangchainOpenaiEmbeddingMaterializer(BaseMaterializer):
+class LangchainOpenaiEmbeddingMaterializer(DefaultMaterializer):
```

Comment:
Can this still read the old files written by `pickle` using `cloudpickle`?

Reply:
Great question. Previously that didn't work because the Python file was not there but I fixed it now. Fortunately, it seems `pickle` and `cloudpickle` are fully compatible otherwise (I tried it with several classes incl. sklearn manually and also added a unit test for this).